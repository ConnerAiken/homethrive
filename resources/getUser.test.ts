const getUser = require('./../resources/getUser');

describe('Get user actions GET->/users/{id}', () => {
    const baseRequest = {
        resource: '/users/{id}',
        httpMethod: 'GET',
    };

    test('We are able to get a user using the GET endpoint', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users/1',
            pathParameters: {
                id: '1',
            },
        };

        const responsePayload = await getUser(requestPayload, {
            get: jest.fn().mockReturnValue({
                promise: () => {
                    return {
                        Item: {
                            id: '1',
                            firstName: 'Conner',
                            lastName: 'Aiken',
                            email: 'conner@fittedtech.com',
                        },
                    };
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(200);
        expect(responsePayload.payload).toMatchSnapshot();
    });

    test('We send the proper server status code when there is no record', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users/1',
            pathParameters: {
                id: '1',
            },
        };

        const responsePayload = await getUser(requestPayload, {
            get: jest.fn().mockReturnValue({
                promise: () => {
                    return {};
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(404);
        expect(responsePayload.body).toContain('Could not find a user');
        expect(responsePayload.body).toMatchSnapshot();
    });

    test('We send the proper server status code when there is missing information', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users/1',
            pathParameters: {},
        };

        const responsePayload = await getUser(requestPayload, {
            get: jest.fn().mockReturnValue({
                promise: () => {
                    return {};
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(400);
        expect(responsePayload.body).toContain('Invalid path parameter');
        expect(responsePayload.body).toMatchSnapshot();
    });

    test('We handle generic errors properly', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users/1',
            pathParameters: {
                id: '1',
            },
        };

        const responsePayload = await getUser(requestPayload, {
            get: jest.fn().mockReturnValue({
                promise: () => {
                    throw new Error('Simulated error');
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(500);
        expect(responsePayload.body).toContain('An error occured');
        expect(responsePayload.body).toMatchSnapshot();
    });
});
