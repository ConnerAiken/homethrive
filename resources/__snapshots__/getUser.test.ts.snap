// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get user actions GET->/users/{id} We are able to get a user using the GET endpoint 1`] = `undefined`;

exports[`Get user actions GET->/users/{id} We handle generic errors properly 1`] = `"{"message":"An error occured","payload":{},"error":{}}"`;

exports[`Get user actions GET->/users/{id} We send the proper server status code when there is missing information 1`] = `"{"error":"Invalid path parameter"}"`;

exports[`Get user actions GET->/users/{id} We send the proper server status code when there is no record 1`] = `"{"message":"Could not find a user with the provided id","error":true,"payload":{}}"`;
